src/app/cli/src/coda.ml

```diff
@@ -9,6 +9,123 @@ open Signature_lib
 open Init
 module YJ = Yojson.Safe
 
+let retrieve_genesis_state dir_opt ~logger :
+    (Ledger.t lazy_t * Proof.t) Deferred.t =
+  let open Cache_dir in
+  let tar_filename = Cache_dir.genesis_dir_name ^ ".tar.gz" in
+  let s3_bucket_prefix =
+    "https://s3-us-west-2.amazonaws.com/snark-keys.o1test.net" ^/ tar_filename
+  in
+  let extract dir =
+    match%map
+      Monitor.try_with_or_error ~extract_exn:true (fun () ->
+          let genesis_dir = dir ^/ Cache_dir.genesis_dir_name in
+          if Core.Sys.file_exists genesis_dir = `Yes then Deferred.return ()
+          else
+            (*Look for the tar and extract*)
+            let tar_file = genesis_dir ^ ".tar.gz" in
+            let%map _result =
+              Process.run_exn ~prog:"tar"
+                ~args:["-C"; dir; "-xzf"; tar_file]
+                ()
+            in
+            () )
+    with
+    | Ok () ->
+        ()
+    | Error e ->
+        Logger.debug ~module_:__MODULE__ ~location:__LOC__ logger
```

Comment:
`error`?

Reply:
This is when looking for the tar in one directory. It's `Fatal` when none of the directories have the file